### DISCUSSION TITLE
Formik + Form Usage - How much of page to wrap ?

### DISCUSSION THREAD

**Discussion by Justinohallo** [jaredpalmer/formik#3405] [General]
I have a large HTML page that contains 90% text and a small form at the bottom. The page is wrapped in a container that has a submit button that is used to submit the form. The submit button cannot be a child of the form, and will always be a parent. 

Is it best practice to wrap the entire page in a Formik + Form component? Or should I create a small form at the bottom of the page, wrap the Inputs in Formik + Form and use pass a ref to the container that has the save button ?

**Comment by johnrom** [comment]
This is generally a bad idea especially from an accessibility point of view. The submit button should be within the context of the form, or else the user may not know what they are submitting or why.

However, for example, you might have a terms of service with 10000 lines of text, and a checkbox at the bottom. In that case the whole thing _is_ a form (submission of the form requires the text), and having a _disabled_ submit button fixed at the bottom could indicate to a user that they need to scroll down to submit. In this case, you can just wrap the whole thing in Formik.

If you really need a submit button outside of a form, you can use the ID of the form to connect a submit button anywhere on the page to it:

```tsx
const MyForm = () => (
  <Formik {...formikProps}>
    <Form id="my-form">
    </Form>
  </Formik>
);

const SomewhereElse = () => (
  <input type="submit" form="my-form" />
)
```

Or you can go the ref direction:

```tsx
const SubmittableFormikContext = createContext({
  formRef: createRef<HTMLFormElement>(null);
});

const MyForm = () => {
  const formRef = useContext(SubmittableFormikContext).formRef;

  return <Formik {...formikConfig} >
    <Form ref={formRef} />
  </Formik>
}

const TelepathicSubmit = () => {
  const formRef = useContext(SubmittableFormikContext).formRef;

  return <button onClick={e => formRef.current?.submitForm()}>Submit the other form</button>
}

const MyApp = () => {
  const formRef = useRef<HTMLFormElement>(null);

  return <SubmittableFormikContext.Provider value={{ formRef }}>
     <MyForm />
     <TelepathicSubmit />
  </SubmittableFormikContext>
}
```
